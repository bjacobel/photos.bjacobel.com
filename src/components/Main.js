import React, { Component } from 'react';
import S3 from 'aws-sdk/clients/s3'; // eslint-disable-line import/order
import AWS from 'aws-sdk/global'; // eslint-disable-line import/order

import styles from '../stylesheets/main.css';
import { COGNITO_POOL_ID, COGNITO_POOL_ROLE_ARN } from '../constants';

export default class Main extends Component {
  constructor() {
    super();
    this.state = {
      photos: [],
    };
  }
  componentWillMount() {
    const {
      match: {
        params: { album },
      },
    } = this.props;
    const prefix = `photos/${album || ''}`;

    AWS.Config.credentials = new AWS.CognitoIdentityCredentials(
      {
        RoleArn: COGNITO_POOL_ROLE_ARN,
        IdentityPoolId: COGNITO_POOL_ID,
      },
      { region: 'us-east-1' }
    );

    AWS.Config.credentials.refreshPromise().then(() => {
      const s3client = new S3({
        credentials: AWS.Config.credentials.webIdentityCredentials,
      });
      s3client
        .listObjectsV2({
          Prefix: prefix,
          Bucket: 'photos.bjacobel.com',
        })
        .promise()
        .then(data => {
          this.setState({
            photos: data.Contents.sort(
              (photo1, photo2) =>
                new Date(photo1.LastModified).getTime() - new Date(photo2.LastModified).getTime()
            ),
          });
        });
    });
  }

  render() {
    return (
      <div className={styles.wall}>
        {this.state.photos.map(photo => (
          <img
            key={photo.ETag + photo.LastModified}
            alt=""
            className={styles.brick}
            src={`https://photos.bjacobel.com/${photo.Key}`}
          />
        ))}
      </div>
    );
  }
}
