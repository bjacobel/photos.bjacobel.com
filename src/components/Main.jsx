import React, { Component } from 'react';
import { connect } from 'react-redux';
import Gallery from 'react-photo-gallery';

import { getPhotosAsync } from '../actions/photos.js';
import Error from './Error';

const mapStateToProps = (state) => {
  return {
    photos: state.photos
  };
};

const mapDispatchToProps = {
  getPhotosAsync
};

class Main extends Component {
  componentWillMount() {
    this.props.getPhotosAsync(this.props.params.collection);
  }

  render() {
    const { photos } = this.props;

    if (photos.hasOwnProperty('error')) {
      return <Error error={ photos.error } />;
    }

    return <Gallery photos={ photos } />;
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Main);
