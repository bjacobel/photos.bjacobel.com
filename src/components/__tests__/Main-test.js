import React from 'react';
import { shallow } from 'enzyme';

import Main from 'components/Main';

describe('main component', () => {
  describe('un-Connected component', () => {
    it('matches snapshot with no params', () => {
      expect(shallow(<Main match={{ params: {} }} />)).toMatchSnapshot();
    });
  });
});
