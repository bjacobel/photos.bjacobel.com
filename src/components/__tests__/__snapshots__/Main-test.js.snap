// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main component un-Connected component matches snapshot with no params 1`] = `
<Query
  query={Object {}}
  variables={
    Object {
      "prefix": "photos/",
    }
  }
/>
`;
